sql: "mismatched types" panic when opening logspy

While running a 4 node local cluster on master (df2e82a76c2ecf7ae445d650ef21461c895b425e) with some local changes unrelated to SQL (zone config proto and allocator changes, to be precise), I enabled logspy on the first node by loading https://localhost:8080/debug/logspy?count=1000&duration=10s&grep=allocator and the node quickly panic'ed with the following logs:

```
W180228 02:15:05.525045 1799 util/log/clog.go:1265  log traffic is now intercepted; log files will be incomplete
I180228 02:15:05.525067 1799 util/log/clog.go:1269  log interception is now stopped; normal logging resumes
I180228 02:15:07.069553 185 server/status/runtime.go:219  [n1] runtime stats: 315 MiB RSS, 213 goroutines, 220 MiB/74 MiB/324 MiB GO alloc/idle/total, 23 MiB/34 MiB CGO alloc/total, 48.49cgo/sec, 0.01/0.00 %(u/s)time, 0.00 %gc (0x)
panic: mismatched types ([{TIMESTAMP 0 0 [] <nil> NONE <nil>} {STRING 0 0 [] <nil> NONE <nil>} {INT 0 0 [] <nil> NONE <nil>} {INT 0 0 [] <nil> NONE <nil>} {STRING 0 0 [] <nil> NONE <nil>} {BYTES 0 0 [] <nil> NONE <nil>}]) and row ([])

goroutine 13952 [running]:
github.com/cockroachdb/cockroach/pkg/sql/sqlbase.EncDatumRow.stringToBuf(0x0, 0x0, 0x0, 0xc428f07e00, 0x6, 0x6, 0xc420abdc80, 0xc42c52acb0)
        /Users/alex/go/src/github.com/cockroachdb/cockroach/pkg/sql/sqlbase/encoded_datum.go:325 +0x2cc
github.com/cockroachdb/cockroach/pkg/sql/sqlbase.EncDatumRow.String(0x0, 0x0, 0x0, 0xc428f07e00, 0x6, 0x6, 0x0, 0xc42caa2910)
        /Users/alex/go/src/github.com/cockroachdb/cockroach/pkg/sql/sqlbase/encoded_datum.go:339 +0xd5
github.com/cockroachdb/cockroach/pkg/sql/distsqlrun.processProducerMessage(0x6ec9b40, 0xc42c050e70, 0x6ed6aa0, 0xc42a3b2ca0, 0x6eb4800, 0xc42c17f380, 0xc429d81c00, 0xc42c4c6b30, 0xc42c7a4e40, 0xc4209a9700, ...)
        /Users/alex/go/src/github.com/cockroachdb/cockroach/pkg/sql/distsqlrun/inbound.go:173 +0x372
github.com/cockroachdb/cockroach/pkg/sql/distsqlrun.processInboundStreamHelper.func2(0xc42dfe0580, 0x6ed6aa0, 0xc42a3b2ca0, 0x6ec9b40, 0xc42c050e70, 0xc42a4bb260, 0xc42c7a4de0, 0x6eb4800, 0xc42c17f380, 0xc429d81c00, ...)
        /Users/alex/go/src/github.com/cockroachdb/cockroach/pkg/sql/distsqlrun/inbound.go:110 +0x11a
created by github.com/cockroachdb/cockroach/pkg/sql/distsqlrun.processInboundStreamHelper
        /Users/alex/go/src/github.com/cockroachdb/cockroach/pkg/sql/distsqlrun/inbound.go:91 +0x1b8
```

There's no load running on the cluster other than having the admin UI open and opening logspy, so I don't know what query even would have been running. I tried to repro by restarting the node and reloading the logspy page, but it works now.

--- a/pkg/sql/distsqlrun/inbound.go
+++ b/pkg/sql/distsqlrun/inbound.go
@@ -166,7 +166,7 @@ func processProducerMessage(
 			return processMessageResult{err: nil, consumerClosed: false}
 		}
 
-		if log.V(3) {
+		if log.V(3) && row != nil {
 			if types == nil {
 				types = sd.Types()
 			}

